src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -3,395 +3,55 @@
 
 using System;
 using System.Collections.Generic;
-using System.IO.Pipelines;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
 using Microsoft.AspNetCore.Hosting.Server;
-using Microsoft.AspNetCore.Hosting.Server.Features;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
-using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
-using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core
 {
+    [Obsolete("This type is obsolete and will be removed in a future version.")]
```

Comment:
This will need an announcement. We should suggest calling IWebHostBuilder.UseKestrel() and resolving the IServer from the built host as an alternative to manually constructing KesetrelServer.

Reply:
Yep, working through first making sure we remove all uses of KestrelServer first 😄